components/infobox/extensions/wikis/starcraft2/infobox_extension_achievements.lua

```diff
@@ -0,0 +1,103 @@
+---
+-- @Liquipedia
+-- wiki=starcraft2
+-- page=Module:Infobox/Extension/Achievements
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local LeagueIcon = require('Module:LeagueIcon')
+local String = require('Module:StringUtils')
+local Team = require('Module:Team')
+
+local Opponent = require('Module:OpponentLibraries').Opponent
+
+local BASE_CONDITIONS = {
+	'[[liquipediatiertype::!Qualifier]]',
+	'[[liquipediatiertype::!Charity]]',
+	'[[liquipediatier::1]]',
+	'[[placement::1]]',
+	'[[prizepoolindex::1]]',
+}
+local GSL_ICON = 'GSLLogo_small.png'
+local GSL_CODE_A_ICON = 'GSL_CodeA.png'
+local CODE_A = 'Code A'
+
+local Achievements = {}
+
+function Achievements.team()
+	local pageName = mw.title.getCurrentTitle().text
+	local historicalPages = Team.queryHistoricalNames(pageName)
+	assert(historicalPages, 'No team template exists for "' .. pageName .. '"')
+	historicalPages = Array.extend(
+		Array.map(historicalPages, function(team) return (team:gsub(' ', '_')) end),
+		Array.map(historicalPages, function(team) return (team:gsub('_', ' ')) end)
+	)
+
+	return Achievements.display(Achievements._fetchDataForTeam(historicalPages, Opponent.team)),
+		Achievements.display(Achievements._fetchDataForTeam(historicalPages, Opponent.solo))
+end
+
+function Achievements._fetchDataForTeam(historicalPages, opponentType)
+	local lpdbKey = opponentType == Opponent.team and 'opponentname' or 'opponentplayers_p1team'
+
+	local conditions = table.concat(Array.extend(
+		BASE_CONDITIONS,
+		'[[opponenttype::' .. opponentType .. ']]',
+		'(' .. table.concat(Array.map(historicalPages, function(team)
+			return '[[' .. lpdbKey .. '::' .. team .. ']]'
+		end), ' OR ') .. ')'
+	), ' AND ')
+
+	return mw.ext.LiquipediaDB.lpdb('placement', {
+		conditions = conditions,
+		query = 'icon, icondark, pagename, shortname, objectname, date, prizepoolindex',
+		order = 'date asc',
+		limit = 5000,
+	})
+end
+
+function Achievements.display(data)
+	if not data or type(data[1]) ~= 'table' then
+		return nil
+	end
+
+	table.sort(data, Achievements._sortByDate)
+
+	return String.nilIfEmpty(table.concat(Array.map(data, Achievements._displayIcon)))
+end
+
+function Achievements._sortByDate(a,b)
+	return a.date < b.date
+end
```

Comment:
Something like a `sortByKey` function could be a nice addition to https://liquipedia.net/commons/Module:Operator, right?

Reply:
#3166